### PR TITLE
Apollo link signInAgainPromise modal

### PR DESCRIPTION
**Pull request by paales**

### CODE REVIEW

**Comment by paales** [comment]
1. Make sure you have an invalid token
2. Go to the product page and add the product to the cart.
3. You now get a the sign in modal
4. Close the modal

I would expect it to show me the unauthorized header.
Except it the request never finishes and the button keeps loading.

File: packages/magento-customer/hooks/useFormIsEmailAvailable.tsx [hunk 1]
```diff
@@ -33,9 +34,14 @@ export function useFormIsEmailAvailable(props: UseFormIsEmailAvailableProps) {
     loggedIn ? 'signedin' : 'email',
   )
 
+  const [, authError] = graphqlErrorByCategory({
+    category: 'graphql-authorization',
+    error: customerQuery.error,
+  })
```
[the remark this answers is not in the document]
Comment (mikekeehnen) [reply packages/magento-customer/hooks/useFormIsEmailAvailable.tsx:40]: `requireAuth` could be false if we didn't invalidate the token yet, so it's probably safer to have this check.

[the remark this answers is not in the document]
Comment (mikekeehnen) [reply packages/magento-customer/hooks/useFormIsEmailAvailable.tsx:40]: I guess you're right

**Comment by mikekeehnen** [comment]
> 1. Make sure you have an invalid token
> 2. Go to the product page and add the product to the cart.
> 3. You now get a the sign in modal
> 4. Close the modal
> 
> I would expect it to show me the unauthorized header. Except it the request never finishes and the button keeps loading.
> 
> So it seems the modal promise never resolves or something?

You were right @paales, we didn't resolve the `pushWithPromise` function in a certain case. This is fixed in my latest commit

**Pull request merged**